kubeflow/core/setup_iap.sh

```diff
@@ -0,0 +1,125 @@
+exit 1
+fi
+
+PROJECT=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/project/project-id)
+if [ -z ${PROJECT} ]; then
+echo Error unable to fetch PROJECT from compute metadata
+exit 1
+fi
+
+PROJECT_NUM=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/project/numeric-project-id)
+if [ -z ${PROJECT_NUM} ]; then
+echo Error unable to fetch PROJECT_NUM from compute metadata
+exit 1
+fi
+
+# Activate the service account
+gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
+# Print out the config for debugging
+gcloud config list
+
+NODE_PORT=$(kubectl --namespace=${NAMESPACE} get svc ${SERVICE} -o jsonpath='{.spec.ports[0].nodePort}')
+while [[ -z ${BACKEND_ID} ]];
+do BACKEND_ID=$(gcloud compute --project=${PROJECT} backend-services list --filter=name~k8s-be-${NODE_PORT}- --format='value(id)');
+echo "Waiting for backend id PROJECT=${PROJECT} NAMESPACE=${NAMESPACE} SERVICE=${SERVICE}...";
+sleep 2;
+done
+echo BACKEND_ID=${BACKEND_ID}
+
+NODE_PORT=$(kubectl --namespace=${NAMESPACE} get svc ${SERVICE} -o jsonpath='{.spec.ports[0].nodePort}')
+BACKEND_SERVICE=$(gcloud --project=${PROJECT} compute backend-services list --filter=name~k8s-be-${NODE_PORT}- --uri)
+# Enable IAP on the backend service:
+gcloud --project=${PROJECT} compute backend-services update ${BACKEND_SERVICE} \
+    --global \
+    --iap=enabled,oauth2-client-id=${CLIENT_ID},oauth2-client-secret=${CLIENT_SECRET}
+
+while [[ -z ${HEALTH_CHECK_URI} ]];
+do HEALTH_CHECK_URI=$(gcloud compute --project=${PROJECT} health-checks list --filter=name~k8s-be-${NODE_PORT}- --uri);
+echo "Waiting for the healthcheck resource PROJECT=${PROJECT} NODEPORT=${NODE_PORT} SERVICE=${SERVICE}...";
+sleep 2;
+done
+
+# Since we create the envoy-ingress ingress object before creating the envoy
+# deployment object, healthcheck will not be configured correctly in the GCP
+# load balancer. It will default the healthcheck request path to a value of
+# / instead of the intended /healthz.
+# Manually update the healthcheck request path to /healthz
+gcloud --project=${PROJECT} compute health-checks update http ${HEALTH_CHECK_URI} --request-path=/healthz
+
+# Since JupyterHub uses websockets we want to increase the backend timeout
+echo Increasing backend timeout for JupyterHub
+gcloud --project=${PROJECT} compute backend-services update --global ${BACKEND_SERVICE} --timeout=3600
+
+JWT_AUDIENCE="/projects/${PROJECT_NUM}/global/backendServices/${BACKEND_ID}"
+
+# For healthcheck compare.
+mkdir -p /var/shared
+echo "JWT_AUDIENCE=${JWT_AUDIENCE}" > /var/shared/healthz.env
+echo "NODE_PORT=${NODE_PORT}" >> /var/shared/healthz.env
+echo "BACKEND_ID=${BACKEND_ID}" >> /var/shared/healthz.env
+
+# TODO(jlewi): We should publish the modified envoy config as a config map
+# and use that in the envoy sidecars.
+kubectl get configmap -n ${NAMESPACE} envoy-config -o jsonpath='{.data.envoy-config\.json}' | \
```

Comment:
No need to write out the envoy-config.json to the shared directory since there is no envoy sidecar in the "iap-enabler" deployment.